services/download/src/main/java/app/coronawarn/server/services/download/FederationGatewayDownloadService.java

```diff
@@ -0,0 +1,106 @@
+/*-
+ * ---license-start
+ * Corona-Warn-App
+ * ---
+ * Copyright (C) 2020 SAP SE and all other contributors
+ * ---
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ---license-end
+ */
+
+package app.coronawarn.server.services.download;
+
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
+
+import app.coronawarn.server.common.federation.client.FederationGatewayClient;
+import app.coronawarn.server.common.protocols.external.exposurenotification.DiagnosisKeyBatch;
+import feign.FeignException;
+import java.time.LocalDate;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+
+/**
+ * Responsible for downloading and storing batch information from the federation gateway.
+ */
+@Component
+public class FederationGatewayDownloadService {
```

Comment:
The only comment i have is that you should consider to move this class to the common federation module.  Other wise all clients of this common module would need to reproduce this wrapper logic over the feign client in order for download to work properly. Either that, or the other way around, move any abstractions that are "higher" than the actual Feign client to the Download service module (like BatchDownloadResponse class <- not referenced anymore in the common module)

Reply:
The first option I don't see much value. Downloading batches is only in the domain of the download service. So I don't see any other consumers using this. But you raise a good point on the validity of having download specific code (i.e. the FeignClient) in the commons module in the first place. The second point I think this would be a bigger refactor/discussion and outside the scope of this PR. Because it would mean falling back to the bare version of the federation commons module. Where the only implementation in commons/federation was the Feign Client Configuration and even the interface was responsibility of each service. I guess that's still a valid approach as we see that there is only a small part of the EFGS integration needs to be shared.

Reply:
The first option I don't see much value. Downloading batches is only in the domain of the download service. So I don't see any other consumers using this. But you raise a good point on the validity of having download specific code (i.e. the FeignClient) in the commons module in the first place. The second point I think this would be a bigger refactor/discussion and outside the scope of this PR. Because it would mean falling back to the bare version of the federation commons module. Where the only implementation in commons/federation was the Feign Client Configuration and even the interface was responsibility of each service. I guess that's still a valid approach as we see that there is only a small part of the EFGS integration needs to be shared.

Reply:
Eugen and me talked yesterday a bit about this topic. BatchDownloadResponse is already moved from the commons module to the downloader. Good points with the shared parts of the FeignClant. Let's keep the discussion for the refactoring for the future.